### PR TITLE
Add review data model

### PR DESCRIPTION
**Pull request by ronan-f**

### CODE REVIEW

File: prisma/migrations/20210710225536_added_reviews/migration.sql [hunk 1]
```diff
@@ -0,0 +1,15 @@
+-- CreateTable
+CREATE TABLE "Review" (
+    "id" SERIAL NOT NULL,
+    "movie_id" TEXT NOT NULL,
+    "author_id" INTEGER NOT NULL,
+    "content" TEXT,
+    "rating" DOUBLE PRECISION NOT NULL,
```
Comment (mostlyfabulous) [inline prisma/migrations/20210710225536_added_reviews/migration.sql:7]: Looks like I can get an accurate representation for 5/7 with this!

File: prisma/migrations/20210710225536_added_reviews/migration.sql [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    "created_at" TIMESTAMP(3) NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    "updated_at" TIMESTAMP(3) NOT NULL,
+
+    PRIMARY KEY ("id")
+);
+
+-- AddForeignKey
+ALTER TABLE "Review" ADD FOREIGN KEY ("author_id") REFERENCES "User"("id") ON DELETE CASCADE ON UPDATE CASCADE;
```
Comment (mostlyfabulous) [inline prisma/migrations/20210710225536_added_reviews/migration.sql:15]: So if a user is deleted, all Reviews by the `author_id` == `id` will also be deleted right?

File: prisma/schema.prisma [hunk 3]
```diff
@@ -8,6 +8,18 @@ generator client {
 }
 
 model User {
-  id    Int    @id @default(autoincrement())
-  email String
+  id     Int      @id @default(autoincrement())
```
Comment (mostlyfabulous) [inline prisma/schema.prisma]: Is it safe when users have a predictable numerical id associated with them? I think GUIDs might be the fashion now days 🤔?

File: prisma/schema.prisma [hunk 4]
```diff
@@ -8,6 +8,18 @@ generator client {
...
+  email  String
+  Review Review[]
+}
+
+model Review {
+  id         Int      @id @default(autoincrement())
+  movie_id   String
+  author_id  Int
+  content    String?
```
Comment (mostlyfabulous) [inline prisma/schema.prisma:20]: Cool, so a Review can be just rating only right?

File: prisma/migrations/20210710225536_added_reviews/migration.sql (hunk 1 above)
Comment (ronan-f) [reply prisma/migrations/20210710225536_added_reviews/migration.sql:7]: lol. I was thinking there will be some validation before adding a row to restrict values to a range of 0-5 with increments of 0.5 but seems like at the database level this is enough?

File: prisma/migrations/20210710225536_added_reviews/migration.sql (hunk 2 above)
Comment (ronan-f) [reply prisma/migrations/20210710225536_added_reviews/migration.sql:15]: Yip that's correct. I would imagine we don't want any orphan reviews? i.e. every review should be owned by a valid user

File: prisma/migrations/20210710225536_added_reviews/migration.sql (hunk 1 above)
Comment (mostlyfabulous) [reply prisma/migrations/20210710225536_added_reviews/migration.sql:7]: Yep!

File: prisma/schema.prisma (hunk 4 above)
Comment (ronan-f) [reply prisma/schema.prisma:20]: Correct...no requirement to actually add text

File: prisma/migrations/20210710225536_added_reviews/migration.sql (hunk 2 above)
Comment (mostlyfabulous) [reply prisma/migrations/20210710225536_added_reviews/migration.sql:15]: Good data purging practice. Respect.

File: prisma/schema.prisma (hunk 3 above)
Comment (ronan-f) [reply prisma/schema.prisma]: Good question. My understanding of the pros of using a GUID would be:

- Unique across multiple DBs/servers
- If a full user object is used client side, GUIDs give bad actors less info

Would you suggest using a GUID as the PK here?

Comment (mostlyfabulous) [reply prisma/schema.prisma]: If you're up for the challenge, yes.

**Review by mostlyfabulous — APPROVED** [review]

File: prisma/schema.prisma (hunk 3 above)
Comment (ronan-f) [reply prisma/schema.prisma]: Pretty easy change with prisma

File: prisma/schema.prisma [hunk 5]
```diff
@@ -8,15 +8,15 @@ generator client {
 }
 
 model User {
-  id     Int      @id @default(autoincrement())
+  id     String   @id @default(uuid())
```
Comment (mostlyfabulous) [inline prisma/schema.prisma:11]: Nice!

**Pull request merged**